sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -158,6 +159,24 @@ object GpuCast {
     "required range"
 
   val INVALID_FLOAT_CAST_MSG = "At least one value is either null or is an invalid number"
+
+  /**
+   * Sanitization step for CAST string to date. Inserts a single zero before any
+   * single digit month or day.
+   */
+  def sanitizeStringToDate(input: ColumnVector): ColumnVector = {
+    val rules = Seq(
+      /// replace yyyy-m with yyyy-mm
+      RegexReplace("(\\A\\d{4})-(\\d{1})\\Z", "\\1-0\\2"),
+      /// replace yyyy-m- with yyyy-mm-
+      RegexReplace("(\\A\\d{4})-(\\d{1}-)", "\\1-0\\2"),
```

Comment:
Is it possible to combine these?  I'm thinking of something similar to what the regex just below does to handle the different endings, for example (also using `raw` strings for improved legibility):
```suggestion
      RegexReplace(raw"(\A\d{4})-(\d{1})([\-|\Z])", raw"\1-0\2\3"),
```

Reply:
Thanks, I was able to combine the first two. Also switched to raw strings.